lib/Frontend/Frontend.cpp

```diff
@@ -424,14 +424,17 @@ void CompilerInstance::setUpDiagnosticOptions() {
 //    Ideally, we'd get rid of it.
 // 2. MemoryBufferSerializedModuleLoader: This is used by LLDB, because it might
 //    already have the module available in memory.
-// 3. ModuleInterfaceLoader: Tries to find an up-to-date swiftmodule. If it
+// 3. ExplicitSwiftModuleLoader: Loads a serialized module if it can, provided
+//    this modules was specified as an explicit input to the compiler.
+// 4. ModuleInterfaceLoader: Tries to find an up-to-date swiftmodule. If it
 //    succeeds, it issues a particular "error" (see
-//    [Note: ModuleInterfaceLoader-defer-to-SerializedModuleLoader]), which
+//    [Note: ModuleInterfaceLoader-defer-to-ImplicitSerializedModuleLoader]), which
 //    is interpreted by the overarching loader as a command to use the
-//    SerializedModuleLoader. If we failed to find a .swiftmodule, this falls
+//    ImplicitSerializedModuleLoader. If we failed to find a .swiftmodule, this falls
```

Comment:
Please be careful about updating comments. There are a couple of issues here:

1. This is exceeding the line length limit now.
2. This broke a reference to the note, since the reference now has a new name but the note's name in `ModuleInterfaceLoader.cpp` is not changed in this PR.

I'm going to fix these two issues in an upcoming PR, it's not a big deal; just something to be careful about.

Reply:
PR with fix: https://github.com/apple/swift/pull/33124